nx/lib/nx/defn/expr.ex

```diff
@@ -211,10 +211,10 @@ defmodule Nx.Defn.Expr do
 
     out =
       if vectorized_axes == [] do
-        out
+        last
```

Comment:
This change is making the `out` lose its shape :(

Reply:
I think the failures are due to us now requiring the initial arg to be properly broadcasted for while. If I go on main and add just the check, I get 33 failures.